tools/builders/packager/build.ts

```diff
@@ -0,0 +1,151 @@
+/**
+ * @license
+ * Copyright 2019 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  BuilderOutput,
+  BuilderContext,
+  Target,
+} from '@angular-devkit/architect';
+
+import { writeFileSync } from 'fs';
+import { join } from 'path';
+import { NgPackagrBuilderOptions } from '@angular-devkit/build-ng-packagr';
+import {
+  tryJsonParse,
+  NgPackagerJson,
+  AngularJson,
+  PackageJson,
+} from '../util/json-utils';
+import { syncPeerDependencyPlaceholder } from './sync-dependencies';
+import { PackagerOptions } from './schema';
+import { copyStyles, copyAssets } from './copy-assets';
+
+export async function packager(
+  userOptions: PackagerOptions,
+  context: BuilderContext,
+): Promise<BuilderOutput> {
+  const options: PackagerOptions = {
+    releasePackageJson: 'package.json',
+    placeholder: '0.0.0-NG',
+    styleFolders: ['style'],
+    assetFolders: ['assets'],
+    ...userOptions,
+  };
+
+  const project = context.target !== undefined ? context.target!.project : '';
+  context.logger.info(`Packaging ${project}...`);
+  const target: Target = {
+    target: 'build',
+    project,
+  };
+
+  // Read the options of the build target options set up with the ng-packagr builder
+  const ngPackagrBuilderOptions = ((await context.getTargetOptions(
+    target,
+  )) as unknown) as NgPackagrBuilderOptions;
+
+  if (
+    ngPackagrBuilderOptions === undefined ||
+    ngPackagrBuilderOptions.project === undefined
+  ) {
+    // If we cannot find the project set for the ng-packagr we cannot continue
+    context.logger.error(
+      'Error: Build target does not have a project set for the ng-packagr in angular.json',
+    );
+    return { success: false };
+  }
+
+  try {
+    const ngPackagrConfigPath = join(
+      context.workspaceRoot,
+      ngPackagrBuilderOptions.project.toString(),
```

Comment:
NIT: `toString()` of a string is maybe not the best option. If the types do not match typecast it in the case when the `project` property changes to an object it would output `"[object Object]"` and the builder fails.